app/forms/questionnaire_form.py

```diff
@@ -37,10 +42,26 @@ def validate(self):
 
         for question in self.schema.get_questions_for_block(self.block_json):
             if question['type'] == 'DateRange':
-                period_from_id = question['answers'][0]['id']
-                period_to_id = question['answers'][1]['id']
-                if not (self.answers_all_valid([period_from_id, period_to_id]) and
-                        self._validate_date_range_question(question['id'], period_from_id, period_to_id)):
+                date_from = question['answers'][0]
+                date_to = question['answers'][1]
```

Comment:
There is quite a bit of code in this if statement now. It may be worth considering moving it into a `self.validate_daterange_question(.....)` method as it does below for calculation and returning the result.